lib/src/clojure_lsp/internal_api.clj

```diff
@@ -251,27 +226,35 @@
   (and ns-exclude-regex
        (re-matches ns-exclude-regex (str namespace))))
 
-(defn ^:private options->namespaces [{:keys [namespace filenames project-root] :as options} db]
-  (or (seq namespace)
-      (->> filenames
-           (map (fn [^File filename-or-dir]
-                  (if (.isAbsolute filename-or-dir)
-                    (io/file filename-or-dir)
-                    (io/file project-root filename-or-dir))))
-           (map (fn [^File filename-or-dir]
-                  (if (shared/directory? filename-or-dir)
-                    (->> filename-or-dir
-                         file-seq
-                         (remove shared/directory?)
-                         (map #(shared/filename->namespace (.getCanonicalPath ^File %) db)))
-                    (shared/filename->namespace (.getCanonicalPath filename-or-dir) db))))
-           flatten
-           (remove nil?)
-           (map symbol)
-           seq)
-      (into #{}
-            (remove (partial exclude-ns? options))
-            (dep-graph/internal-ns-names db))))
+(defn ^:private options->uris [{:keys [namespace filenames project-root] :as options} db]
```

Comment:
The api accepts filenames or namespaces as a specification for what to clean or report on. As a fallback, it operates on the whole project. The code for figuring out what to do used to convert everything to namespaces, then find matching filenames. Now we convert everything to URIs first, and report on what namespaces were operated on afterward. This removes a lot of duplication and tricky code.